extensions/ql-vscode/src/queries.ts

```diff
@@ -273,7 +280,13 @@ async function checkAndConfirmDatabaseUpgrade(qs: qsClient.QueryServerClient, db
 
   logger.log(descriptionMessage);
   // Ask the user to confirm the upgrade.
-  const shouldUpgrade = await helpers.showBinaryChoiceDialog(`Should the database ${db.databaseUri.fsPath} be upgraded?\n\n${descriptionMessage}`);
+
+  let messageLines = descriptionMessage.split('\n');
+  if (messageLines.length > MAX_MESSAGE_LINES) {
+    messageLines = messageLines.slice(0, MAX_MESSAGE_LINES);
+    messageLines.push("... [full list of changes can be found in Output > CodeQL Extension Log]");
```

Comment:
It might be helpful here to explain a little more about how to get to the CodeQL Extension Log, for example like we do on compilation error messages.  Another option could be adding a button that focuses the log.

Reply:
Yeah, a button to cancel-but-show-log sounds like a good idea.